integration/stability/soak_parallel_rm.sh

```diff
@@ -0,0 +1,168 @@
+#!/bin/bash
+# Copyright (c) 2017 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# This test will run a number of parallel containers, and then try to
+# 'rm -f' them all at the same time. It will check after each run and
+# rm that we have the expected number of containers, shims, proxys,
+# qemus and runtimes active
+# The goals are two fold:
+# - spot any stuck or non-started components
+# - catch any hang ups
+
+# How many times will we run the test loop...
+ITERATIONS=5
+
+# the system 'free available' level where we stop running the tests, as otherwise
+#  the system can crawl to a halt, and/or start refusing to launch new VMs anyway
+# We choose 2G, as that is one of the default VM sizes for CC
+MEM_CUTOFF=(2*1024*1024*1024)
+
+# The default container/workload we run for testing
+# nginx has show up issues in the past, for whatever reason, so
+# let's default to that
+PAYLOAD="nginx"
+
+# do we need a command argument for this payload?
+COMMAND=""
+
+# Set the runtime if not set already
+RUNTIME="${RUNTIME:-cc-runtime}"
+
+# And set the names of the processes we look for
+QEMU_NAME="${QEMU_NAME:-qemu-lite-system-x86_64}"
+RUNTIME_NAME="${RUNTIME_NAME:-cc-runtime}"
+SHIM_NAME="${SHIM_NAME:-cc-shim}"
+PROXY_NAME="${PROXY_NAME:-cc-proxy}"
+
+# let's cap the test. If you want to run until you hit the memory limit
+# then just set this to a very large number
+MAX_CONTAINERS=110
+
+function count_containers() {
+	docker ps -qa | wc -l
+}
+
+function check_all_running() {
+	echo Checking ${how_many} containers have all relevant components
+
+	# check what docker thinks
+	how_many_running=$(count_containers)
+
+	if (( ${how_many_running} != ${how_many} )); then
+		echo "Wrong number of containers running (${how_many_running} != ${how_many}) - stopping"
+		exit -1
+	fi
+
+	# check we have the right number of proxy's
+	how_many_proxys=$(ps --no-header -C ${PROXY_NAME} | wc -l)
+	if (( ${how_many_running} >= 1 )); then
+		# If we have any containers, then we expect to have a single proxy
+		if (( ${how_many_proxys} != 1 )); then
```

Comment:
if this tests is designed for `runc` as a RUNTIME, could be `${how_many_proxys}` > 1, this is due to `runc` does not use cc-proxy or something like that, then `how_many_proxys` will be 0 and this an expected result

Reply:
Indeed. Right now the primary goal is to run with CC, but if we were to support 'other' runtimes, then yes, we need to make the script more sentient.
I'm probably about to code something similar up in another script - I'll see if I get to apply that here as well.